### PR TITLE
rebase: update sidecars to latest versions

### PR DESCRIPTION
**Pull request by humblec**
external-provisioner: v2.3.0
external-attacher: v3.3.0
external-snapshotter: v4.2.0
node-driver-registrar: v2.3.0

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Review by nixpanic — DISMISSED** [review]
Update `build.env` as well please.

**Comment by humblec** [comment]
@Mergifyio rebase

**Comment by humblec** [comment]
#2362 
```
Aug 17 05:45:16.645: INFO: attach command running after restart, runningAttachCmd: rbd-nbd --id cephcsi-rbd-node -m rook-ceph-mon-a.rook-ceph.svc.cluster.local:6789 --keyfile=/tmp/csi/keys/keyfile-788080743 attach replicapool/csi-vol-ccd171f7-ff15-11eb-888e-e2bb86bddc97 --device /dev/nbd0 --try-netlink --reattach-timeout=300
rbd-nbd --id cephcsi-rbd-node -m rook-ceph-mon-a.rook-ceph.svc.cluster.local:6789 --keyfile=/tmp/csi/keys/keyfile-788080743 attach replicapool/csi-vol-ccd171f7-ff15-11eb-888e-e2bb86bddc97 --device /dev/nbd0 --try-netlink --reattach-timeout=300

Aug 17 05:45:16.655: INFO: ExecWithOptions {Command:[/bin/sh -c echo 'Hello World' > /var/lib/www/html/test] Namespace:rbd-694 PodName:csi-rbd-demo-pod ContainerName:web-server Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
Aug 17 05:45:16.655: INFO: >>> kubeConfig: /root/.kube/config
Aug 17 05:45:19.527: INFO: stdErr occurred: /bin/sh: 1: cannot create /var/lib/www/html/test: Read-only file system

Aug 17 05:45:19.527: FAIL: failed to write IO, err: command terminated with exit code 2, stdErr: /bin/sh: 1: cannot create /var/lib/www/html/test: Read-only file system
 ```

File: charts/ceph-csi-rbd/values.yaml [hunk 1]
```diff
@@ -207,7 +207,7 @@ provisioner:
     enabled: true
     image:
       repository: k8s.gcr.io/sig-storage/csi-attacher
-      tag: v3.2.1
+      tag: v3.3.0
```
Comment (Madhu-1) [inline charts/ceph-csi-rbd/values.yaml:210]: csi-provisioner version is not updated for rbd

File: charts/ceph-csi-cephfs/values.yaml [hunk 2]
```diff
@@ -161,7 +161,7 @@ provisioner:
   provisioner:
     image:
       repository: k8s.gcr.io/sig-storage/csi-provisioner
-      tag: v2.2.2
+      tag: v3.0.0
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/values.yaml:164]: node-driver-registrar is not updated for cephfs

**Review by Madhu-1 — DISMISSED** [review]

File: charts/ceph-csi-cephfs/values.yaml (hunk 2 above)
Comment (humblec) [reply charts/ceph-csi-cephfs/values.yaml:164]: Thanks @Madhu-1 , updated .

File: charts/ceph-csi-rbd/values.yaml (hunk 1 above)
Comment (humblec) [reply charts/ceph-csi-rbd/values.yaml:210]: updated.. thanks .ptal @Madhu-1

**Review by Madhu-1 — APPROVED** [review]

**Review by nixpanic — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Comment by yati1998** [comment]
> /retest ci/centos/mini-e2e-helm/k8s-1.20

Failed with error: `    Aug 17 08:57:47.155: failed to write IO, err: command terminated with exit code 2, stdErr: /bin/sh: 1: cannot create /var/lib/www/html/test: Read-only file system`. [logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e-helm_k8s-1.20/detail/mini-e2e-helm_k8s-1.20/2116/pipeline
)

**Pull request merged**